### DISCUSSION TITLE
GitHub Pages and npm

### DISCUSSION THREAD

**Discussion by vitboch** [community/community#31172] [Other Feature Feedback, Questions, & Ideas] [answered]
Hello! I wonder if it's possible to publish a page on GitHub that requires npm to run?

**Comment by jfullstackdev** [comment]
what do you mean by that ? 

you want to run your npm on GitHub Pages directly ? 

GitHub Pages only runs HTML, CSS and JavaScript directly, if there is any developer that can run npm directly using JavaScript, that may be possible

**Comment by brean** [comment, marked as answer]
Yes, you can use github actions to run npm to build a static page and deploy it to github pages:
https://github.com/marketplace/actions/deploy-to-github-pages
